File: tests/fixtures/sample.ts
import { EventEmitter } from 'events';

interface Greetable {
    greet(name: string): string;
}

class Animal {
    constructor(public name: string) {}

    public speak(): string {
        return `${this.name} makes a sound.`;
    }
}

class Dog extends Animal implements Greetable {
    private breed: string;

    constructor(name: string, breed: string) {
        super(name);
        this.breed = breed;
    }

    public speak(): string {
        return `${this.name} barks.`;
    }

    public greet(person: string): string {
        return `Woof! Hello, ${person}!`;
    }

    private fetchBreed(): string {
        return this.breed;
    }
}

function formatAnimal(animal: Animal): string {
    return `[${animal.name}]`;
}

File: tests/fixtures/sample.py
import os
from pathlib import Path

class Animal:
    def __init__(self, name: str):
        self.name = name

    def speak(self) -> str:
        return f"{self.name} makes a sound."

class Dog(Animal):
    def __init__(self, name: str, breed: str):
        super().__init__(name)
        self.breed = breed

    def speak(self) -> str:
        return f"{self.name} barks."

    def fetch_breed(self) -> str:
        return self.breed

def format_animal(animal: Animal) -> str:
    return f"[{animal.name}]"
